pilot/cmd/pilot-agent/main.go

```diff
@@ -90,6 +90,7 @@ var (
 	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
 	istioNamespaceVar    = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
 	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	clusterIDVar         = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
```

Comment:
Don't we have it in proxy config already ? 

I think @howardjohn moved almost everything to use ProxyConfig - with proper annotation to override, etc.

At this point we shouldn't add more env variables - at least not for important things like cluster ID.

Reply:
I'm almost certain we have some other ways to pass the cluster ID ( or registry ID actually ), please check with @howardjohn - he fixed some related bug. Sorry, my memory is bad.

Reply:
This is not a new env.

Reply:
Hi @hzxuzhonghu and @costinm 
I found a ClusterID at "pilot/pkg/model/context.go"(type Proxy struct) with comment:
`
type Proxy struct {
	// ClusterID specifies the cluster where the proxy resides.
	// TODO: clarify if this is needed in the new 'network' model, likely needs to
	// be renamed to 'network'
	ClusterID string
`
It's seems that model contains all meta related to the proxy network (include ClusterID)

Reply:
@hovonako That one in the NodeMeta indicates the cluster for proxy, here it is where the istiod resides.